juju/juju
# 2.1.1 fix network choosing 1665648

This is a couple of small tweaks from Horacio's Issue #7030 

## Description of change

One issue that we are addressing is that we were just using 10.0.X.0/24 and starting at 0 and just walking up until we found a value. Now we are walking randomly so 2 machines are less likely to collide.
The other factor is that the old algorithm assumed all networks were /24, so if it encountered a 10.0.3.0/23 then it had a chance to use 10.0.4.0/24 which is actually still in use.

## QA steps

You can bootstrap to somewhere like AWS, and then ssh into that machine, create bridges/assign IP addresses for a bunch of 10.0.* address ranges. Then do `juju add-machine lxd:0`. We should avoid all of the ranges that you've already specified.

## Documentation changes

I don't believe we documented that lxdbr0 address ranges will always be the next available 10.0.*.0/24 range, which gives us freedom to change the algorithm without changing what we were offering.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1665648
This also touches on:
https://bugs.launchpad.net/juju/+bug/1657850
It doesn't do everything asked there, but it does improve the situation because every machine won't try to choose exactly the same prefix.
